demos/python_demos/common/performance_metrics.py

```diff
@@ -0,0 +1,86 @@
+"""
+ Copyright (C) 2020 Intel Corporation
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+from time import perf_counter
+import cv2
+from helpers import put_highlighted_text
+
+
+class Sum:
+    def __init__(self):
+        self.latency = 0.0
+        self.time = 0.0
+        self.frame_count = 0
+
+
+class PerformanceMetrics:
+    def __init__(self, time_window=1.0):    # defines the length of the timespan used for 'current fps' calculation,
+        self.time_window_size = time_window # set to '1 second' by default
+        self.last_moving_sum = Sum()
+        self.current_moving_sum = Sum()
+        self.total_sum = Sum()
+        self.last_update_time = perf_counter()
+    
+    def update(self, last_request_start_time):
+        current_time = perf_counter()
+
+        last_request_latency = current_time - last_request_start_time
+        self.current_moving_sum.latency += last_request_latency
+        self.current_moving_sum.frame_count += 1
+        if self.current_moving_sum.frame_count == 1 and self.total_sum.frame_count == 0:
+            self.last_moving_sum.latency = last_request_latency
+            self.last_moving_sum.frame_count = 1
+
+        if current_time - self.last_update_time > self.time_window_size:
+            self.current_moving_sum.time = current_time - self.last_update_time
+            self.last_update_time = current_time
+            self.last_moving_sum = self.current_moving_sum
+            self.current_moving_sum = Sum()
+            self.total_sum.latency += self.last_moving_sum.latency
+            self.total_sum.time += self.last_moving_sum.time
+            self.total_sum.frame_count += self.last_moving_sum.frame_count
+
+    def get_current_fps(self):
```

Comment:
There is no reason to separate fps and latency: if you needed fps and latency to be kept separate you wouldn't count them in a single `update()` method.

It even makes sense to merge `update()` and `show_current_*()` methods (which actually should have been called `put_moving_*()`). I can't come up with a scenario when you don't call `show_current_*()` after `update()` except for that use case with Tab in your implementation. Which has a problem: it does't draw latency for a while after switching to min_latency mode with nireq set big enough. It will be better if you find a way to track previous and current mode more accurate in the demo.

Reply:
Actually demos that use console can't make use of `PerformanceMetrics`. Usually we don't want to keep moving average for them at all.

Reply:
> Actually demos that use console can't make use of `PerformanceMetrics`. Usually we don't want to keep moving average for them at all.

@Wovchena, do you mean that since console demos can't use PerformanceMetrics, we can safely merge update() and  put_moving_*()? Or do you mean that I should add another type of metric specially for console demos?

Reply:
> do you mean that since console demos can't use PerformanceMetrics, we can safely merge update() and put_moving_*()?

Yes.

But if you see a nice way to modify/split `PerformanceMetrics` to use it in console only demos, we can consider that approach. For example a base class for console demos and its derivative to report moving stats.

On the other hand metric computing for console demos shall be very simple and it can be easier to use bare `perf_counter()` without introducing any helper.

The idea behind merging methods is to prevent forgetting to call one of the methods as they are always expected to be called one after another in common use case. And to prevent calling that methods far form each other like
```
performanceMetrics.update()
postprocess_detections()
performanceMetrics.put_moving_*()
```